force-app/main/domain/Households.cls

```diff
@@ -99,20 +99,50 @@ public inherited sharing class Households extends fflib_SObjects2 {
         set;
     }
 
+    public Households(){}
+
     /*******************************************************************************************************
     * @description Constructor
     * @param accountIds Account Ids to initialize the Contact Repository with
     */
-    public Households(Set<Id> accountIds) {
+    @TestVisible
+    private Households(Set<Id> accountIds) {
         super((List<Object>) householdSelector.findByIds(accountIds));
     }
 
     /*******************************************************************************************************
     * @description Constructor
     * @param accountIds Account Ids to initialize the Contact Repository with
     */
-    public Households(List<Account> accounts) {
-        super(accounts);
+    @TestVisible
+    private Households(List<SObject> accounts) {
+        super(accounts, Account.SObjectType);
+    }
+
+    public static IHouseholds newInstance(List<SObject> records) {
+        Application.Domain.replaceWith(Account.SObjectType, IHouseholds.class, Households.Constructor.class);
+        return (IHouseholds) Application.Domain.newInstance(records, Account.SObjectType);
+    }
+
+    public static IHouseholds newInstance(Set<Id> recordIds) {
+        Application.Domain.replaceWith(Account.SObjectType, IHouseholds.class, Households.Constructor.class);
+        return (IHouseholds) Application.Domain.newInstance(householdSelector.findByIds(recordIds), Account.SObjectType);
+    }
+
+    public List<SObject> getHouseholds() {
+        return getRecords(isHouseholdAccount());
+    }
+
+    private fflib_Criteria isHouseholdAccount() {
```

Comment:
It's important to call out that this method will be used to filter to only Household accounts anywhere where we use `newInstance()` to get an instance of `Household` records in a calling class. We need to make sure this filter criteria is accurate to define what a "Household" account actually is. @force2b wanted to get your input on this.

Reply:
Is the question here the filtering or the reasoning why it has to filter (which I'm not totally clear on)?